Sources/Core/Microsoft.StreamProcessing/Egress/AtemporalArray/AtemporalArrayEgressPipe.cs

```diff
@@ -111,6 +111,23 @@ private void EnqueueDelete(long currentTime, TPayload payload)
             queue.Add(payload);
         }
 
+        public override void OnCompleted()
+        {
+            OnFlush();
+            this.observer.OnCompleted();
```

Comment:
Should this call base.OnCompleted, not call on the observer directly?

Reply:
Yes

---
In reply to: [241596174](https://github.com/Microsoft/Trill/pull/18#discussion_r241596174) [](ancestors = 241596174)